tactics/rewrite.ml

```diff
@@ -1612,15 +1624,7 @@ let cl_rewrite_clause_newtac ?abs ?origsigma ~progress strat clause =
     | None -> concl
     | Some id -> EConstr.of_constr (Environ.named_type id env)
     in
-    let env = match clause with
-    | None -> env
-    | Some id ->
-      (* Only consider variables not depending on [id] *)
-      let ctx = named_context env in
-      let filter decl = not (occur_var_in_decl env sigma id decl) in
```

Comment:
This filter removes variables depending on `id` but not `id` itself, if you fix that (without other changes) it seems to work
~~~ocaml
      let filter decl = not (Id.equal id (NamedDecl.get_id decl)) && not (occur_var_in_decl env sigma id decl) in
~~~
(also update the comment 2 lines above)

Reply:
Isn't that exactly what this PR is doing?

Reply:
Note that the other changes are needed to make `rewrite` work. Otherwise, only `setoid_rewrite` is fixed.

Reply:
>Isn't that exactly what this PR is doing?

yes, I misread